src/core/symbol/SymbolTransaction.ts

```diff
@@ -0,0 +1,39 @@
+import { Converter, KeyPair, SymbolNetwork, SymbolTransactionUtils } from '@core';
+import { TransactionBuilder, TransactionHelper } from 'catbuffer-typescript';
+
+export interface SignedTransaction {
+    payload: Uint8Array;
+    signature: Uint8Array;
+    transactionHash: string;
+}
+
+export interface TransactionSigner {
+    signTransaction(payload: Uint8Array, generationHashBytes: Uint8Array): Promise<SignedTransaction>;
+}
+
+export class KeyParTransactionSigner implements TransactionSigner {
```

Comment:
do we absolutely need the `TransactionSigner`? Can pass in KeyPair and then keyPair.sign(..) perhaps?

Reply:
Please have a read at this comment https://github.com/symbol/sdk-typescript/issues/9#issuecomment-887423254

The WIP idea is that the generic SymbolTransaction can be used with private key, Ledger, and Trezor accounts. The problem with the current SDK is that people need to copy and paste logic because SDK assumes that there is always a private key when signing.

Reply:
So, I had a second look and because all the low level operations are in this `SymbolTransactionUtils`, the ledger/trezor implementation could talk to them if required, for example `SymbolTransactionUtils.addCosignatures` and `SymbolTransactionUtils.addSignature`. These ops don't need a private key as the signing part is done in other operations.

I've removed the TransactionSigner and async methods from the SymbolTransaction. This class is still WIP. Need to write down more examples, play with mosaic/namespaces ids and aggregates.